FluxC: Fix post list progress bar getting stuck

oguzkocer opened the pull request:
Fixes #5135. This PR fixes the mentioned issue with a reset however since we don't get the `onPostChanged` event, it's not possible to set the `mCanLoadMorePosts` flag. That shouldn't be a big deal though, once we do another fetch, it will be set to `false` if needed.

To test:
* Go into post list
* Trigger a manual fetch and immediately go into post detail
* Return back to post list and verify the progress bar is not visible anymore

maxme commented:
So I just took a look on this and I think it makes more sense to move the dispatcher registration to onCreate on unregistration to onDestroy, like it was done in `SelectCategoriesActivity`.

I double checked and we need to do that in `PostsListFragment` but also in `CommentsListFragment`, these are the only places where this can happen.

oguzkocer commented:
@maxime I've reverted the previous solution and moved the register/unregister to create/destroy actions. I've fixed the Comment one here as well as they are pretty much the same bug. We could update the issue title if it's a concern though.

maxme commented on WordPress/src/main/java/org/wordpress/android/ui/comments/CommentsListFragment.java:
```diff
@@ -104,17 +104,12 @@ public CommentStatus toCommentStatus() {
-    public void onStop() {
+    public void onDestroy() {
         mDispatcher.unregister(this);
         super.onStop();
```
`super.onDestroy();` must be called here

oguzkocer replied:
:facepalm: Sorry! Not good enough of an excuse, but I am practically sleep walking today :(

maxme commented:
:shipit:
